src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1323,17 +1329,37 @@ export class NetworkAreaDiagramViewer {
         }
     }
 
+    private getCssPropertyValue(displayedSize: number, propertyValue: ((value: number) => string) | string): string {
+        if (typeof propertyValue === 'function') {
+            return propertyValue(displayedSize);
+        }
+        return propertyValue;
+    }
+
     public initializeDynamicCssRules(maxDisplayedSize: number) {
         this.getDynamicCssRules().forEach((rule) => {
-            rule.thresholdStatus = maxDisplayedSize < rule.threshold ? THRESHOLD_STATUS.BELOW : THRESHOLD_STATUS.ABOVE;
+            switch (rule.type) {
+                case CSS_RULE_TYPE.THRESHOLD_DRIVEN: {
+                    rule.thresholdStatus =
+                        maxDisplayedSize < rule.threshold ? THRESHOLD_STATUS.BELOW : THRESHOLD_STATUS.ABOVE;
+                    break;
+                }
+                case CSS_RULE_TYPE.FUNCTION_DRIVEN: {
+                    for (const [property, propertyValue] of Object.entries(rule.cssDeclaration)) {
+                        rule.currentValue[property] = this.getCssPropertyValue(maxDisplayedSize, propertyValue);
```

Comment:
propertyValue should be called callbackFunction

Reply:
done